### DISCUSSION TITLE
How to get last element of array?

### DISCUSSION THREAD

**Discussion by Noxor** [community/community#25673] [Actions]
<p>I found an  <a href="https://github.community/t/referecing-an-element-from-array-in-yml/17042">issue</a> explaining how to access element of array by index. Can reverse indexes be used as well, or is there any way to access last element? </p>
<pre><code>run: echo '${{github.event.commits[-1].message}}'
</code></pre>
<p>returns empty string even through when</p>
<pre><code>run: echo '${{ toJson(github) }}'
</code></pre>
<p>is called, there is a text in the message of last commit</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/noxor">@noxor</a> ,</p>
<p>To get the github context element, please check the json body firstly and follow the structure for your code. For instance, my ‘on: push’ event github context as below:</p>
<p><img src="https://user-images.githubusercontent.com/52772761/181092454-f4c4e8c8-87d2-4d7c-857b-9c212e140a40.png" alt="commitcheck.png" data-base62-sha1="mnBPZD6IUsnE3P6fEnJQ8qST2r1" width="688" height="476"></p>
<p>commits body has only one element which highlighted with {}, you can get the message from ${{ github.event.commits[0].message }}.</p>
<p>And there’s a ‘head_commit’ element, you can get the head commit(last) message from </p>
<p>${{ github.event.head_commit.message }} as well.</p>
<p>If the body has mutliple elements like below array structure(fake just for example), you can use <span class="math">{{ github.event.commits[0].message }} for message 1, and&amp;nbsp;</span>{{ github.event.commits[1].message }}  for message 2.</p>
<pre><code>{
....
"commits": [
      {
        "message": "this is message 1"
      },
      {
        "message2": "this is message 2"
      }
    ],
...
}
</code></pre>
<p>Hope it’s clear and helpful!</p>

**Comment by Noxor** [comment]
<p>Thanks for reply,</p>
<p>the ‘head_commit’ seems to solve my problem, since I needed to read message of merge commit to release branch. It was not possible to use normal indexes, because number of elements in ‘commits’ can vary.</p>

**Comment by merlinstardust** [comment]
<p>While <code>head_commit</code> does solve the intended issue, it doesn’t answer the original question.</p>
<p>How do you get the last element of an array?</p>
<p>Doing <code>commits[1]</code> is fine but how do you determine what number you should use to get the last element?</p>
<p>Does accessing arrays in JSON in YAML have a <code>commits.lastEntry</code> or a <code>commits.length</code> or something?</p>
